### ISSUE TITLE
`pixi add` a command line app from within a pixi shell is not immediately available

### ISSUE THREAD

**Issue by synapticarbors** [prefix-dev/pixi#977]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://github.com/prefix-dev/pixi/releases) of pixi, using `pixi --version`.

### Reproducible example

```shell
❯ pixi init test1
✔ Initialized project in /Users/xyz/opt/pixienvs/test1
❯ which rg
/Users/xyz/.pixi/bin/rg
❯ rg --version
ripgrep 14.1.0

features:-simd-accel,+pcre2
simd(compile):+SSE2,+SSSE3,-AVX2
simd(runtime):+SSE2,+SSSE3,+AVX2

PCRE2 10.42 is available (JIT is available)
❯ cd test1
❯ pixi shell
 . "/var/folders/ld/gx_9s3_n55g1dkhj59zjlxww0000gn/T/pixi_env_OJF.sh"
❯  . "/var/folders/ld/gx_9s3_n55g1dkhj59zjlxww0000gn/T/pixi_env_OJF.sh"
❯ which rg
/Users/xyz/.pixi/bin/rg
❯ pixi add "ripgrep=14.0.3"
✔ Added ripgrep=14.0.3
❯ which rg
/Users/xyz/.pixi/bin/rg
❯ ls -l .pixi/envs/default/bin/rg
-rwxrwxr-x  1 xyz  staff  5142352 Dec  4 06:44 .pixi/envs/default/bin/rg
❯ rg --version
ripgrep 14.1.0

features:-simd-accel,+pcre2
simd(compile):+SSE2,+SSSE3,-AVX2
simd(runtime):+SSE2,+SSSE3,+AVX2

PCRE2 10.42 is available (JIT is available)
❯ exit
❯ pixi shell
 . "/var/folders/ld/gx_9s3_n55g1dkhj59zjlxww0000gn/T/pixi_env_J4t.sh"
❯  . "/var/folders/ld/gx_9s3_n55g1dkhj59zjlxww0000gn/T/pixi_env_J4t.sh"
❯ which rg
/Users/xyz/opt/pixienvs/test1/.pixi/envs/default/bin/rg
❯ rg --version
ripgrep 14.0.3

features:-simd-accel,+pcre2
simd(compile):+SSE2,+SSSE3,-AVX2
simd(runtime):+SSE2,+SSSE3,+AVX2

PCRE2 10.42 is available (JIT is available)
```

### Issue description

If you `pixi add` a command line tool (in this example ripgrep), while in a pixi shell for that project, the tool is not immediately available even though it's on `$PATH`. In the example I gave above, I have ripgrep installed in a global environment, and that continues to be one that's available after I get a different version using `pixi add`, even though it should take precedence in the path, which is:

```
echo $PATH
/Users/xyz/opt/pixienvs/test1/.pixi/envs/default/bin:...<lots of other stuff>...:/Users/xyz/.pixi/bin
``` 

I have to exit the `pixi shell` and then restart it before I get access to the one that is in that environment. 

I discovered this with `pip`, when I was trying to do an editable install. I had forgotten to add `pip` to my environment, so when I was in the `pixi shell`, I ran `pixi add pip` and then `pip install -e .`, and the package ended up in a different environment (actually a conda environment that I had activated before starting the shell). 

### Expected behavior

If I'm in a `pixi shell` and add a command line tool, it should be immediately available since it's on the `$PATH` in a `bin` directory with highest precedence.

**Comment by wolfv** [comment]
This is a duplicate of https://github.com/prefix-dev/pixi/issues/326 

I think we need to add a "hook" to call re-source the shell (or re-hash) after an installation, indeed.

**Comment by wolfv** [comment]
I am closing this for now as it should be (partially) fixed with the next release. I've also added a new issue for a more complete fix here: https://github.com/prefix-dev/pixi/issues/1018
